components/FilterItem.js

```diff
@@ -248,8 +253,14 @@ const FilterItem = forwardRef((props, ref) => {
 					className='flex border-[6px] items-center justify-center rounded-full'
 					onPress={() => {
 						props.toggleShown();
-						updateFilter(props.data.id);
-						updateFilterName(props.data.title);
+						setFilter(props.data.id);
+						setFilterName(props.data.title);
+						setFilterIcon(props.data.icon);
+						setFilterColor(props.data.color);
+						setFilterDescription(props.data.description); //NRA
+						setFilterIconColor(props.data.iconColor);
+						setCircleMembersData(props.data.circleMembersData)
+						console.log(props.data.circleMembersData);
 					}}
```

Comment:
based on my above comment, this should not be stored globally since it doesn't need to be accessed globally. rather we can just use a local state in `circleSettings` and update this everytime the current circle id is changed. let me know if you need help with this

Reply:
[done]